contrib/test/integration/main.yml

```diff
@@ -59,16 +60,24 @@
   remote_user: root
   vars_files:
     - "{{ playbook_dir }}/vars.yml"
-  tags:
-    - integration
   tasks:
     - name: clone build and install cri-tools
       include: "build/cri-tools.yml"
       vars:
         force_clone: True
         cri_tools_git_version: "9da2549287cbbdbfa045b3246ab34ffea703dce4"
+      tags:
+        - integration
+    - name: Usernamespace integration testing needs updated environment variables
+      set_fact:
+        # Keys/Values in userns_int_test_env update int_test_env (from vars.yml)
+        int_test_env: '{{ int_test_env | combine(userns_int_test_env) }}'
+      tags:
+        - userns
```

Comment:
I have a reminder for me regarding tags in ansible: DO NOT USE THEM. They're useless, counterintuitive, misleading, evil. That little reminder comes in handy every time I think I want to try using tags.

And I still don't think this does what you think it does.

Reply:
They have their place, and yes they're confusing.  They work like this:

Internally, depending on the ``strategy`` used, essentially the entire playbook gets "flattened" into into a single, giant task list.  Any tasks not selected (by tag or skip-tag) is removed from the list. 
Lastly, Ansible runs the task list against the inventory.

When flattening, conditionals (``when:``) and tags combine downwards, from play -> role -> include -> block -> task.  The thing I forgot is...tags are all logically "or" combined, while conditions use ``and``.  

Hope that helps somewhat.

---

Unfortunately, tags are here and in wide-spread and long-term use by lower layers.  It doesn't make sense to re-write everything at this point, so...lemons to lemon-aid.

For the usernamespace testing jobs, the playbook gets called here: https://github.com/openshift/aos-cd-jobs/pull/1409/files#diff-42b680df8e935f400e7249da605ab349R24  

So tasks matching either ``integration`` ***OR*** ``userns`` are selected.   Best I can tell, what I did here should work, since I removed the integration tag from the play level.

It's possible to duplicate the OriginCI job definition, push that up into a Jenkins job, and fire that job against this specific PR.  That should prove if it works or not :grinning:   BTW- instructions for doing that, are here:  https://github.com/openshift/aos-cd-jobs/blob/master/README.md#custom-xml-generator-configuration-under-sjb